superwall/src/main/java/com/superwall/sdk/config/ConfigManager.kt

```diff
@@ -347,6 +348,59 @@ open class ConfigManager(
         }
     }
 
+    internal suspend fun refreshConfiguration() {
+        // Make sure config already exists
+        val oldConfig = config ?: return
+
+        // Ensure the config refresh feature flag is enabled
+        if (!oldConfig.featureFlags.enableConfigRefresh) {
+            return
+        }
+
+        try {
+            val newConfig =
+                network.getConfig {
+                    configState.update { Result.Success(ConfigState.Retrying) }
+                }
+            removeUnusedPaywallVCsFromCache(oldConfig, newConfig)
+            processConfig(newConfig)
+            configState.update { Result.Success(ConfigState.Retrieved(newConfig)) }
+            Superwall.instance.track(InternalSuperwallEvent.ConfigRefresh)
+            ioScope.launch { preloadPaywalls() }
+        } catch (e: Exception) {
+            configState.update { Result.Success(ConfigState.Retrieved(oldConfig)) }
+            Logger.debug(
+                logLevel = LogLevel.warn,
+                scope = LogScope.superwallCore,
+                message = "Failed to refresh configuration.",
+                info = null,
+                error = e,
+            )
+        }
+    }
+
+    private suspend fun removeUnusedPaywallVCsFromCache(
+        oldConfig: Config,
+        newConfig: Config,
+    ) {
+        val oldPaywallIds = oldConfig.paywalls.map { it.identifier }.toSet()
+        val newPaywallIds = newConfig.paywalls.map { it.identifier }.toSet()
+
+        val presentedPaywallId = paywallManager.currentView?.paywall?.identifier
+        val missingPaywallIds =
+            if (presentedPaywallId != null) {
+                oldPaywallIds.minus(presentedPaywallId).subtract(newPaywallIds)
+            } else {
+                oldPaywallIds.subtract(newPaywallIds)
+            }
+
+        missingPaywallIds.forEach {
+            paywallManager.removePaywallView(it)
```

Comment:
You'll need to create the cache key and then remove the paywall based on its cache key, rather than ID